File: bsc_be/bsc-be/bsc-be/DTOs/TransactionResponse.cs
using bsc_be.Models;

namespace bsc_be.DTOs
{
    public class TransactionResponse
    {
        public long Id { get; set; }
        public string GigName { get; set; } = string.Empty;
        public string TransactionStatus { get; set; } = string.Empty;
        public decimal BasePrice { get; set; }
        public decimal TotalPriceReceived { get; set; }
        public decimal TotalPricePaid { get; set; }
        public string BuyerDescription { get; set; } = string.Empty;
        public ItemResponse? Item { get; set; }
        public RatingResponse? Rating { get; set; }
        public DateTime Date { get; set; }
        public DateTime Deadline { get; set;}
    }
}

File: bsc_fe/bsc-fe/src/app/interfaces/transaction-response.ts
import { TransactionStatus } from "../enums/transaction-status";
import { Item } from "./item";
import { Rating } from "./rating";

export interface TransactionResponse {
    id: number;
    gigName: string;
    transactionStatus: TransactionStatus;
    totalPrice: number;
    date: Date;
    rating: Rating;
    item: Item;
}
